bin/build-artifacts-in-docker.sh

```diff
@@ -1,37 +1,98 @@
+                              script's current directory.
+  [--base_image_name <name>]: Name of Docker image to use during the build. Defaults to prestodev/centos6-presto-admin-tests.
+  [--base_image_tag <tag>]:   Tag of the Docker image to use during the build. Defaults to the tag stored in the base_images_tag.json file.
+  [--online-dist]:            Boolean argument indicating whether the online distribution should be build. If neither the
+                              --online-dist nor the --offline-dist flags are specified then both distributions are built.
+  [--offline-dist]:           Boolean argument indicating whether the offline distribution should be build. If neither the
+                              --online-dist nor the --offline-dist flags are specified then both distributions are built.
+EOF
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+  case $key in
+      --root_dir)
+      ROOT_DIR="$2"
+      shift
+      shift
+      ;;
+      --base_image_name)
+      BASE_IMAGE_NAME="$2"
+      shift
+      shift
+      ;;
+      --base_image_tag)
+      BASE_IMAGE_TAG="$2"
+      shift
+      shift
+      ;;
+      --online-dist)
+      DIST_TARGETS="$DIST_TARGETS dist-online "
+      shift
+      ;;
+      --offline-dist)
+      DIST_TARGETS="$DIST_TARGETS dist-offline "
+      shift
+      ;;
+      *)    # unknown; show usage and error out
+      echo "Unknown option $key"
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${ROOT_DIR}" ]]; then
+  ROOT_DIR=$(readlink -f $(dirname $0)/..)
+fi
 
 if [[ -z "${BASE_IMAGE_NAME}" ]]; then
   BASE_IMAGE_NAME="prestodev/centos6-presto-admin-tests"
 fi
 
-BASE_IMAGE_NAME=${BASE_IMAGE_NAME}-build
-
 if [[ -z "${BASE_IMAGE_TAG}" ]]; then
   BASE_IMAGE_TAG=$(cat ${ROOT_DIR}/base-images-tag.json | python -c 'import sys, json; print json.load(sys.stdin)["base_images_tag"]')
 fi
 
+if [[ -z "${DIST_TARGETS}" ]]; then
+  DIST_TARGETS="dist-online dist-offline"
+fi
+
+echo "ROOT_DIR        = ${ROOT_DIR}"
+echo "BASE_IMAGE_NAME = ${BASE_IMAGE_NAME}"
+echo "BASE_IMAGE_TAG  = ${BASE_IMAGE_TAG}"
+echo "DIST_TARGETS    = ${DIST_TARGETS}"
+
+BASE_IMAGE_NAME=${BASE_IMAGE_NAME}-build
+
 echo Building presto-admin-artifacts in container ${BASE_IMAGE_NAME}:${BASE_IMAGE_TAG}
 
 CONTAINER_NAME="presto-admin-build-$(date '+%s')"
 CONTAINER_DIR="/mnt/presto-admin"
 
 docker run --name ${CONTAINER_NAME} -v ${ROOT_DIR}:${CONTAINER_DIR} --rm -i ${BASE_IMAGE_NAME}:${BASE_IMAGE_TAG} \
-  env CONTAINER_DIR="${CONTAINER_DIR}" bash <<"EOF"
+  env CONTAINER_DIR="${CONTAINER_DIR}" env DIST_TARGETS="${DIST_TARGETS}" bash <<"EOF"
     set -x
     cd ${CONTAINER_DIR}
-    pip install --ignore-installed -r requirements.txt
-    pip install tox-travis==0.10
-    # list presto-admin package dependencies versions
-    pip install Babel==2.5.3
-    pip install cffi==1.11.5
-    pip install PyNaCl==1.2.1
-    pip install cryptography==2.1.1
```

Comment:
why did you remove requirements.txt ?

Reply:
`requirements.txt` is used for setting up your development env so it contains dependencies that are not relevant for building the installer.